### PR TITLE
Re-enable a few more StyleCop rules (1019, 1020, 1112, 1113, 1115, 1302, 1517)

### PR DESCRIPTION
**Pull request by stephentoub**
Filling the few-minutes gaps between several meetings today...

### CODE REVIEW

File: src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs [hunk 1]
```diff
@@ -1393,10 +1393,10 @@ private async Task InternalSendRequestBodyAsync(WinHttpRequestState state, bool
             using (var requestStream = new WinHttpRequestStream(state, chunkedModeForSend))
             {
                 await state.RequestMessage.Content.CopyToAsync(
-                    requestStream,
-                    state.TransportContext
 #if HTTP_DLL
-                    , state.CancellationToken
```
Comment (sharwell) [inline src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs]: 💭 It almost seems like this should be an exception, but I'm OK with the change as well.

File: src/System.Private.Xml/src/System/Xml/Xsl/XsltOld/RecordOutput.cs [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 namespace System.Xml.Xsl.XsltOld
 {
-    internal interface RecordOutput
+    internal interface IRecordOutput
```
Comment (sharwell) [inline src/System.Private.Xml/src/System/Xml/Xsl/XsltOld/RecordOutput.cs]: 💡 Rename file to match?

File: src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs (hunk 1 above)
Comment (stephentoub) [reply src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs]: > It almost seems like this should be an exception, but I'm OK with the change as well.

My preference is actually that the rule takes into account ifdefs like this (can you make that happen? :wink:).  There are other rules that have this issue as well, to the extent that we won't be enabling them for now.

For this specific case, I went back and forth what to do about it, but decided the readability both ways was equally clear, and didn't want to make it more convoluted with adding pragma warning disables and restores, nor a SuppressMessage attribute.

File: src/System.Private.Xml/src/System/Xml/Xsl/XsltOld/RecordOutput.cs (hunk 2 above)
Comment (stephentoub) [reply src/System.Private.Xml/src/System/Xml/Xsl/XsltOld/RecordOutput.cs]: Yup, meant to do that. Thanks.

**Pull request merged**